model_analyzer/analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -0,0 +1,108 @@
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from itertools import product
+from subprocess import check_output, CalledProcessError, STDOUT
+
+
+class PerfAnalyzer:
+    """A wrapper class for the perf_analyzer. This
+    class provides an interface for running workloads 
+    with perf_analyzer
+    """
+
+    def __init__(self, config):
+        """
+        Parameters
+        ----------
+        config : PerfAnalyzerConfig
+            keys are names of arguments to perf_analyzer,
+            values are their values.
+        """
+        self._config = config
+        self._outputs = []
+
+    def run_job(self, sweep_params):
+        """Parses the parameters to sweep over and runs
+        the perf_analyzer once per configuration.
+
+        Parameters
+        ----------
+        sweep_params : dictionary of lists
+            keys are arguments to perf_analyzer, values are 
+            list of argument values to run perf_analyzer with.
+        
+        Returns
+        -------
+        List of tuples 
+            (params, output) where params are the set of 
+            parameters provided for that run and output is 
+            the stdout from the perf_analyzer.
+        """
+
+        # Create a config for each combination of parameters
+        combos = list(product(*tuple(sweep_params.values())))
+        run_configs = [dict(zip(sweep_params.keys(), vals)) for vals in combos]
+
+        # Collect a tuple (run_config, output)
+        run_outputs = []
+
+        # Run parameters are some subset of perf_analyzer config
+        for run_config in run_configs:
+
+            # update perf_analyzer config
+            for key, val in run_config.items():
+                self._config[key] = val
+            
+            # Run perf_analyzer
+            run_output = self._run_perf_analyzer()
+            run_outputs.append((run_config, run_output))
+        
+        return run_outputs
+
+    def _run_perf_analyzer(self):
+        """Runs the perf_analyzer tool locally 
+        and synchronously
+        
+        Returns
+        -------
+        str
+            output from the subprocess call to 
+            the perf_analyzer
+
+        """
+        cmd = ['perf_analyzer']
+        cmd += self._config.to_cli_string().replace('=', ' ').split()
+
+        # Synchronously start and finish run
+
+        try:    
+            out = check_output(cmd, stderr=STDOUT, encoding='utf-8')
+        except CalledProcessError as e:
+            raise Exception(f"perf analyzer returned with exit" 
```

Comment:
What about creating `TritonModelAnalyzerException` class so that this can be reused across the application?

Reply:
Ah yes. Good idea thank you.